packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,267 @@
+
+  /**
+   * Whether the checkbox should be rendered as checked by default.
+   */
+  defaultChecked?: 'indeterminate' | boolean;
+
+  /**
+   * Checkbox supports two different checkbox sizes.
+   * @defaultvalue 'medium'
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Determines whether the label should be positioned before (start) or after (end) the checkbox.
+   * @defaultvalue 'end'
+   */
+  labelPosition?: 'start' | 'end';
+
+  /**
+   * ID of the root element that wraps the checkbox and label.
+   */
+  rootId?: string;
+
+  /**
+   * ID of the <input/> element that represents the checkbox.
+   */
+  id?: string;
+
+  /**
+   * Callback to be called when the checked state value changes.
+   */
+  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: CheckboxOnChangeData) => void;
+}
+
+/**
+ * Data for the onChange event for checkbox.
+ */
+export interface CheckboxOnChangeData {
+  checked?: 'indeterminate' | boolean;
+}
+```
+
+Some feedback needed for props are:
+
+- `checked` will work similar to `aria-checked` and have `true`, `false`, and `indeterminate` options.
+- A `rootId` is needed since the input and label will share an id and `rootId` will be used for the root element.
+- To initially make the checkbox simple, the icon will be a `ReactElement` instead of a slot.
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### **Public**
 
-## Migration
+```tsx
+<Checkbox label="Example Checkbox" />
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+### **Internal**
 
-- _Migration from v8_
-- _Migration from v0_
+```tsx
+<slots.root {...slotProps.root}>
+  {state.labelPosition === 'start' && <slots.label {...slotProps.label} />}
+  <div className={state.checkboxClassName}>
+    <div className={state.iconClassName}>{icon}</div>
+    <input
+      type="checkbox"
+      className={state.inputClassName}
+      ref={state.inputRef}
+      id={state.inputId}
+      onChange={state.inputOnChange}
+    />
+  </div>
+  {state.labelPosition === 'end' && <slots.label {...slotProps.label} />}
+</slots.root>
+```
 
-## Behaviors
+### **DOM**
+
+In this case, the label position is set to end as per default.
+
+```tsx
+<div {/* Root Element */}>
+  <div {/* Checkbox */}>
+    <div {/* Icon */}>
+      ...Icon
+    </div>
+    <input id="cbox-1" type="checkbox" aria-checked="..." aria-label="..." aria-labelledby="..."
+      aria-disabled="..." aria-posinset="..." aria-setsize="..." />
+  </div>
+  <label for="cbox-1">
+    Example Checkbox
+  </label>
+</div>
+```
 
-_Explain how the component will behave in use, including:_
+## Migration
+
+See [MIGRATION.md](MIGRATION.md)
+
+## Behaviors
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+- Keyboard
+  - When checkbox is focused, if `space` is pressed the checkbox's state is toggled.
```

Comment:
Q: Should `Enter` trigger the same behavior?

Reply:
Let's stay consistent with the WAI keyboard patterns:
* https://www.w3.org/TR/wai-aria-practices-1.2/examples/checkbox/checkbox-2/checkbox-2.html
* https://www.w3.org/TR/wai-aria-practices-1.2/examples/checkbox/checkbox-1/checkbox-1.html